File: app/src/main/java/com/hiteshsahu/awesome_gallery/modal/CenterRepository.java
package com.hiteshsahu.awesome_gallery.modal;

/**
 * Created by Hitesh on 23-07-2016.
 */
public class CenterRepository {

    private ImageCollection imageCollection = new ImageCollection();

    private static CenterRepository centerRepositoryInstance;

    private CenterRepository() {
    }

    public static CenterRepository getInstance() {

        if (null == centerRepositoryInstance) {
            centerRepositoryInstance = new CenterRepository();
        }
        return centerRepositoryInstance;
    }

    public ImageCollection getImageCollection() {
        return imageCollection;
    }


    public void clearImageCollection() {
        imageCollection.getListOfImages().clear();
    }

}

File: app/src/main/java/com/hiteshsahu/awesome_gallery/modal/CenterRepository.kt
package com.hiteshsahu.awesome_gallery.modal

/**
 * Created by Hitesh on 23-07-2016.
 */
class CenterRepository private constructor() {

    val imageCollection = ImageCollection()

    fun clearImageCollection() {
        imageCollection.listOfImages.clear()
    }

    companion object {

        private var centerRepositoryInstance: CenterRepository? = null

        val instance: CenterRepository
            get() {
                if (null == centerRepositoryInstance) {
                    centerRepositoryInstance = CenterRepository()
                }
                return centerRepositoryInstance!!
            }
    }

}
